### PR TITLE
IconName Custom Type

### PR DESCRIPTION
**Pull request by tayambamwanza**
@avmaisak  This was running successfully from my machine, it maybe worth testing before merging on your side though just to be sure, let me know if there are any issues.

I also modified the path to icon folder in generation script, instead of '../icons/icons' I changed to just 'icons', so that the script can be run from the root of the project, I just run, ```node tools/generate-icons```

The icons folder has the icons in the top of the folder, so when unzipping I don't have the folder icons/icons but just icons folder, and all the svgs pasted in there.

### CODE REVIEW

File: projects/ngx-bootstrap-icons-lib/src/lib/components/ngx-bootstrap-icons/ngx-bootstrap-icons.component.ts [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+import { Icons } from "../../providers/icon.provider";
+import { IconName } from "../../types/icon-name.type";
 
 /**
  * Bootstrap icon component.
  */
 @Component({
-  selector: 'i-bs, *[i-bs]',
-  template: '<ng-content></ng-content>',
+  selector: "i-bs, *[i-bs]",
```
Comment (avmaisak) [inline projects/ngx-bootstrap-icons-lib/src/lib/components/ngx-bootstrap-icons/ngx-bootstrap-icons.component.ts:23]: in project uses single quotes

**Comment by tayambamwanza** [comment]
@avmaisak That must've been my auto-formatting, I'll fix that.

File: projects/ngx-bootstrap-icons-lib/src/lib/enums/icon-names.enum.ts [hunk 2]
```diff
@@ -340,6 +340,12 @@ export enum IconNamesEnum {
   Bookshelf = 'bookshelf',
   /** https://icons.getbootstrap.com/icons/bootstrap-fill */
   BootstrapFill = 'bootstrap-fill',
+  /** https://icons.getbootstrap.com/icons/bootstrap-icons.css */
+  BootstrapIconsCss = 'bootstrap-icons.css',
```
Comment (avmaisak) [inline projects/ngx-bootstrap-icons-lib/src/lib/enums/icon-names.enum.ts:344]: what is this?

Comment (tayambamwanza) [reply projects/ngx-bootstrap-icons-lib/src/lib/enums/icon-names.enum.ts:344]: @avmaisak Ah, I didn't notice that, I downloaded the bootstrap icons and then ran the generation script, seems they included the css file, I can delete it and rerun the script.

Comment (tayambamwanza) [reply projects/ngx-bootstrap-icons-lib/src/lib/enums/icon-names.enum.ts]: There was also a json file that I removed now

**Comment by avmaisak** [comment]
@tayambamwanza please, resolve conflicts

**Comment by tayambamwanza** [comment]
> @tayambamwanza please, resolve conflicts

@avmaisak seems to be resolved now.

**Comment by avmaisak** [comment]
> > @tayambamwanza please, resolve conflicts
> 
> @avmaisak seems to be resolved now.

not resolved

**Comment by tayambamwanza** [comment]
@avmaisak I see you merged? are the conflicts resolved? I tried using github editor and you said it didn't show, did it just take time to show? I was just about to go into my local instant to make the commit.

The issue was just that in the icons folder there was a file called index.html.ts

**Comment by avmaisak** [comment]
i fixed it, merged , and released

**Comment by avmaisak** [comment]
@tayambamwanza need readme to be updated, with these feature

**Comment by tayambamwanza** [comment]
@avmaisak Hi, sorry I got really busy, for this feature I don't think it needs to be mentioned in readme, the dev should just see the autocomplete as soon as they start typing an icon name, showing what options are available.

**Pull request merged**